app/firebase/functions/src/notifications/notification.service.ts

```diff
@@ -0,0 +1,210 @@
+  }
+
+  async getFull(
+    userId: string,
+    notificationId: string,
+    manager: TransactionManager
+  ): Promise<NotificationFull> {
+    const notification = await this.notificationsRepo.get(
+      userId,
+      notificationId,
+      manager,
+      true
+    );
+
+    const activity = await this.activityRepo.get(
+      notification.activityId,
+      manager,
+      true
+    );
+
+    return {
+      ...notification,
+      activity,
+    };
+  }
+
+  /** Aggregates all pending notifications of a user and sends them as an email */
+  async notifyUser(userId: string) {
+    await this.db.run(async (manager) => {
+      const user = await this.usersRepo.getUser(userId, manager, true);
+      const settings = user.settings;
+
+      /** get pending notificatations */
+      const pendingIds = await this.notificationsRepo.getUnotifiedOfUser(
+        userId,
+        manager
+      );
+
+      const pendingNotifications = await Promise.all(
+        pendingIds.map((notificationId) =>
+          this.getFull(userId, notificationId, manager)
+        )
+      );
+
+      /** check if user has enabled notififications */
+      if (settings.notificationFreq === NotificationFreq.None) {
+        return;
+      }
+
+      const autopostingOptions = [
+        AutopostOption.AI,
+        AutopostOption.DETERMINISTIC,
+      ];
+
+      if (
+        autopostingOptions.includes(settings.autopost[PLATFORM.Nanopub].value)
+      ) {
+        await this.prepareAndSendDigestAuto(
+          userId,
+          pendingNotifications,
+          manager
+        );
+      } else {
+        await this.prepareAndSendDigestManual(
+          userId,
+          pendingNotifications,
+          manager
+        );
+      }
+
+      await Promise.all(
+        pendingNotifications.map((n) =>
+          this.notificationsRepo.markAsNotified(userId, n.id, manager)
+        )
+      );
+    });
+  }
+
+  async getPostEmailDetails(
+    notification: NotificationFull,
+    types: ActivityType[],
+    manager: TransactionManager
+  ) {
+    if (types.includes(notification.activity.type)) {
+      const post = await this.postsRepo.get(
+        (notification as PostParsedNotification).activity.data.postId,
+        manager,
+        true
+      );
+
+      const postText = PostsHelper.concatenateThread(post);
+
+      return {
+        content: postText,
+        url: getPostUrl(post.id),
+      };
+    } else {
+      throw new Error('Unsupported notification type');
+    }
+  }
+
+  async prepareAndSendDigestManual(
+    userId: string,
+    notifications: NotificationFull[],
+    manager: TransactionManager
+  ) {
+    /** */
+    const postsDetails: EmailPostDetails[] = await Promise.all(
+      notifications.map(async (notification) =>
+        this.getPostEmailDetails(
+          notification,
+          [ActivityType.PostParsed],
+          manager
+        )
+      )
+    );
+
+    await this.sendDigest(userId, postsDetails);
```

Comment:
we'll need to make sure that the email template knows if it is for parsed posts or autoposted posts